devices/Adxl343/ADXL343.cs

```diff
@@ -0,0 +1,1821 @@
+        /// </summary>
+        /// <param name="actAcDc">Active AC/DC.</param>
+        /// <param name="actXEnable">Active X Enable.</param>
+        /// <param name="actYEnable">Active Y Enable.</param>
+        /// <param name="actZEnable">Active Z Enable.</param>
+        /// <param name="inactAcDc">Inactive AC/DC.</param>
+        /// <param name="inactXEnable">Inactive X Enable.</param>
+        /// <param name="inactYEnable">Inactive Y Enable.</param>
+        /// <param name="inactZEnable">Inactive Z Enable.</param>
+        /// <returns>True if Full Transfer result, false if any other result.</returns>
+        public bool SetActiveInactiveControl(bool actAcDc, bool actXEnable, bool actYEnable, bool actZEnable, bool inactAcDc, bool inactXEnable, bool inactYEnable, bool inactZEnable)
+        {
+            byte map = 0;
+
+            if (actAcDc)
+            {
+                map |= (byte)ActInactCtlMap.ActAcDc;
+            }
+
+            if (actXEnable)
+            {
+                map |= (byte)ActInactCtlMap.ActXEnable;
+            }
+
+            if (actYEnable)
+            {
+                map |= (byte)ActInactCtlMap.ActYEnable;
+            }
+
+            if (actZEnable)
+            {
+                map |= (byte)ActInactCtlMap.ActZEnable;
+            }
+
+            if (inactAcDc)
+            {
+                map |= (byte)ActInactCtlMap.InactAcDc;
+            }
+
+            if (inactXEnable)
+            {
+                map |= (byte)ActInactCtlMap.InactXEnable;
+            }
+
+            if (inactYEnable)
+            {
+                map |= (byte)ActInactCtlMap.InactYEnable;
+            }
+
+            if (inactZEnable)
+            {
+                map |= (byte)ActInactCtlMap.InactZEnable;
+            }
+
+            SpanByte writeBuf = new byte[2];
+            writeBuf[0] = (byte)Register.TapAxes;
+            writeBuf[1] = map;
+            var res = i2c.Write(writeBuf);
+
+            if (res.Status == I2cTransferStatus.FullTransfer)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        #endregion
+
+        #region Thresh FF
+
+        /// <summary>
+        /// The THRESH_FF register is eight bits and holds the threshold
+        /// value, in unsigned format, for free-fall detection.The acceleration
+        /// on all axes is compared with the value in THRESH_FF to determine
+        /// if a free-fall event occurred.The scale factor is 62.5 mg/LSB.Note
+        /// that a value of 0 mg may result in undesirable behavior if the
+        /// free-fall interrupt is enabled.Values between 300 mg and 600 mg
+        /// (0x05 to 0x09) are recommended.
+        /// </summary>
+        /// <param name="threshFF">Free-fall detection value.</param>
+        /// <returns>True if Full Transfer result, false if any other result.</returns>
+        public bool GetThreshFF(ref int threshFF)
```

Comment:
Here as well, try pattern, full name rather than abbreviation, checking parameters for the set and possibility property as well or instead of the functions and UnitsNet as it's a gram unit